contracts/strategy/LiquidationStrategy.sol

```diff
@@ -441,6 +501,105 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
         }
     }
 
+    function _sellCollateralV3(
+        address _tokenIn,
+        address _tokenOut,
+        uint256 _amountIn,
+        uint24 _poolFee,
+        address _routerV3
+    ) internal returns (uint256 receivedAmount) {
+        uint256 _tokencoinBalanceBefore = ERC20(_tokenOut).balanceOf(address(this));
+
+        (bytes memory commands, bytes[] memory inputs) = _encodeSwapExactInputSingle(
+            address(this),
+            _amountIn,
+            0,
+            abi.encodePacked(_tokenIn, _poolFee, _tokenOut),
+            true
+        );
+
+        ERC20(_tokenIn).safeApprove(uniswapV3Info[_routerV3].permit2, _amountIn);
+
+        IPermit2(uniswapV3Info[_routerV3].permit2).approve(
+            _tokenIn,
+            address(uniswapV3Info[_routerV3].universalRouter),
+            uint160(_amountIn),
+            uint48(block.timestamp)
+        );
+
+        IUniversalRouter(uniswapV3Info[_routerV3].universalRouter).execute(commands, inputs, block.timestamp);
+
+        ERC20(_tokenIn).safeApprove(uniswapV3Info[_routerV3].permit2, 0);
+
+        uint256 _tokencoinBalanceAfter = ERC20(_tokenOut).balanceOf(address(this));
+
+        receivedAmount = _tokencoinBalanceAfter.sub(_tokencoinBalanceBefore);
+
+        return receivedAmount;
+    }
+
+    // @dev _computeMostProfitablePath should be upgraded/updated once curvePool launches and DEX pool of USDT/FXD will be drained.
+    // an alternative solution can be to involvement of Xswap(UniswapV3)
+    function _computeMostProfitablePath(
+        IUniswapV2Router02 _router,
+        address _collateralToken,
+        uint256 _collateralAmountToLiquidate
+    ) internal view returns (address[] memory, uint256) {
+        // DEX (Collateral -> FXD)
+        address[] memory path1 = new address[](2);
+        path1[0] = _collateralToken;
+        path1[1] = address(fathomStablecoin);
+        uint256 scenarioOneAmountOut = _getDexAmountOut(_collateralAmountToLiquidate, path1, _router);
+
+        // DEX (Collateral -> USDT) -> DEX (USDT -> FXD)
+        address[] memory path2 = new address[](3);
```

Comment:
What is pair collateral-> usdt doesn't exist? In this case we will have a revert here